lib/chai/core/assertions.js

```diff
@@ -1765,6 +1765,12 @@ module.exports = function (chai, _) {
       , obj = flag(this, 'object')
       , ssfi = flag(this, 'ssfi');
 
+    if (typeof name !== 'string' && isNested) {
+      var msgPrefix = flag(this, 'message')
+      msgPrefix = msgPrefix ? msgPrefix + ': ' : ''
```

Comment:
The logic here on line 1770 is repeated on lines 1769 and 1778 (except with `flagMsg`); I'm thinking we can reduce duplication by removing all three of these and instead performing the operation a single time near the top after line 1766 where the variable declarations take place.

Reply:
Yes, there is little duplication of logic here . We can remove msgPrefix and use flagMsg variable in its place and move the `flagMsg = flagMsg ? flagMsg + ': ' : ''`  near the top as you suggested.